src/models/hooks/goalTemplate.js

```diff
@@ -0,0 +1,74 @@
+// import { auditLogger } from '../../logger';
+import { CREATION_METHOD } from '../../constants';
+
+const { Op } = require('sequelize');
+
+const autoPopulateHash = (sequelize, instance) => {
+  const changed = instance.changed();
+  if (Array.isArray(changed)
+    && changed.includes('templateName')
+    && instance.templateName !== null
+    && instance.templateName !== undefined) {
+    instance.set('hash', sequelize.fn('md5', sequelize.fn('TRIM', instance.templateName)));
+  }
+};
+
+const autoPopulateTemplateNameModifiedAt = (sequelize, instance) => {
+  const changed = instance.changed();
+  if (Array.isArray(changed)
+    && changed.includes('templateName')
+    && instance.templateName !== null
+    && instance.templateName !== undefined) {
+    instance.set('templateNameModifiedAt', new Date());
+  }
+};
+
+const autoPopulateCreationMethod = (sequelize, instance) => {
+  const changed = instance.changed();
+  if (Array.isArray(changed)
+        && (!changed.includes('creationMethod')
+        || instance.creationMethod === null
+        || instance.creationMethod === undefined)) {
+    instance.set('creationMethod', CREATION_METHOD[0]); // 'Automatic'
```

Comment:
Could clean this up by defining the constants like

```
export const AUTOMATIC_CREATION = 'automatic'
export const CURATED_CREATION = 'curated'

export const CREATION_METHOD = [AUTOMATIC_CREATION, CURATED_CREATION]
```
that way we could use `AUTOMATIC_CREATION` here instead of the array index (might be better names for the string constants)